riscv/triggers.cc

```diff
@@ -13,6 +13,21 @@ void trigger_with_tdata2_t::tdata2_write(processor_t UNUSED * const proc, const
   tdata2 = val;
 }
 
+reg_t disabled_trigger_t::tdata1_read(const processor_t * const proc) const noexcept
+{
+  auto xlen = proc->get_xlen();
+  reg_t tdata1 = 0;
+  tdata1 = set_field(tdata1, CSR_TDATA1_TYPE(xlen), CSR_TDATA1_TYPE_DISABLED);
+  tdata1 = set_field(tdata1, CSR_TDATA1_DMODE(xlen), dmode);
+  return tdata1;
+}
+
+void disabled_trigger_t::tdata1_write(processor_t * const proc, const reg_t val) noexcept
+{
+  auto xlen = proc->get_xlen();
+  dmode = proc->get_state()->debug_mode ? get_field(val, CSR_TDATA1_DMODE(xlen)) : 0;
```

Comment:
This checking of `debug_mode` becomes obsolete after ca58beb7fb2e737054d61f05937e9a05a1333882